tools/ui-components/src/button/button.tsx

```diff
@@ -65,17 +100,39 @@ export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
       size = 'medium',
       type = 'button',
       onClick,
-      children
+      children,
+      disabled,
+      block
     },
     ref
   ) => {
     const classes = useMemo(
-      () => computeClassNames({ size, variant }),
-      [size, variant]
+      () => computeClassNames({ size, variant, disabled, block }),
+      [size, variant, disabled, block]
+    );
+
+    // Manually prevent click event if the button is disabled
+    // as `aria-disabled` marks the element disabled but still registers the click event.
+    // Ref: https://css-tricks.com/making-disabled-buttons-more-inclusive/#aa-the-difference-between-disabled-and-aria-disabled
+    const handleClick = useCallback(
+      (event: React.MouseEvent<HTMLButtonElement>) => {
+        const ariaDisabled = event.currentTarget.getAttribute('aria-disabled');
+
+        if (!ariaDisabled && onClick) {
+          onClick(event);
+        }
+      },
+      [onClick]
     );
 
     return (
-      <button ref={ref} className={classes} type={type} onClick={onClick}>
+      <button
+        ref={ref}
+        className={classes}
+        type={type}
+        onClick={handleClick}
+        aria-disabled={disabled}
```

Comment:
`aria-disabled` is more preferred as it allows the user to focus on the element and understand that the button is disabled. While with the `disabled` attribute, the button is not focusable, and non-sighted users won't know about its existence nor its disabled state.

Refs: 
- https://www.a11y-101.com/development/aria-disabled
- https://css-tricks.com/making-disabled-buttons-more-inclusive